core/web/router.go

```diff
@@ -384,11 +384,16 @@ func uiCorsHandler(config store.Config) gin.HandlerFunc {
 
 func readBody(reader io.Reader) string {
 	buf := new(bytes.Buffer)
-	buf.ReadFrom(reader)
+	_, err := buf.ReadFrom(reader)
+	if err != nil {
+		logger.Warn("unable to read from body for sanitazation: ", err)
+		return "*FAILED TO READ BODY*"
+	}
 
 	s, err := readSanitizedJSON(buf)
 	if err != nil {
-		return buf.String()
+		logger.Warn("unable to sanitize json for logging: ", err)
+		return "*FAILED TO READ BODY*"
```

Comment:
This looks like a semipredicate error, surely we should change the return to `(string, error)` ?

Reply:
You're right, but as is, it fits nicely in the sole calling function above. Considering this is already merged and not critical, leaving as is.

```
		logger.Infow(fmt.Sprintf("%s %s", c.Request.Method, c.Request.URL.Path),
			"method", c.Request.Method,
			"status", c.Writer.Status(),
			"path", c.Request.URL.Path,
			"query", redact(c.Request.URL.Query()),
			"body", readBody(rdr), // <--- here
```